test/e2e/pipeline/infra/gke/setup.sh

```diff
@@ -0,0 +1,115 @@
+
+function gke_create_cluster() {
+    local cluster_id=$1
+    local cluster_region=$2
+    local cluster_zone=$3
+    local num_nodes=$4
+    local index=$5
+
+    local cluster_version="${K8S_VERSION%.*}"
+
+    local arg_dataplane=""
+    if [[ $GKE_DATAPLANE == "v2" ]]; then
+        arg_dataplane="--enable-dataplane-v2"
+    fi
+
+    pod_cidr=10.${index}.0.0/16
+    if [[ ${POD_CIDR_OVERLAPPING} == "true" ]]; then
+        pod_cidr=10.0.0.0/16
+    fi
+        
+    "${GCLOUD}" container --project "${GCLOUD_PROJECT_ID}" clusters create "${cluster_id}" --zone "${cluster_zone}" \
+        --num-nodes "${num_nodes}" --machine-type "${GKE_MACHINE_TYPE}" --image-type "${GKE_IMAGE_TYPE}" --disk-type "${GKE_DISK_TYPE}" --disk-size "${GKE_DISK_SIZE}" \
+        --cluster-version "${cluster_version}" --no-enable-intra-node-visibility --enable-shielded-nodes --enable-ip-alias \
+        --release-channel "regular" --no-enable-basic-auth --metadata disable-legacy-endpoints=true \
+        --network "projects/${GCLOUD_PROJECT_ID}/global/networks/liqo-${index}" --subnetwork "projects/${GCLOUD_PROJECT_ID}/regions/${cluster_region}/subnetworks/liqo-nodes" $arg_dataplane --cluster-ipv4-cidr="${pod_cidr}" \
+        --default-max-pods-per-node "110" --security-posture=standard --workload-vulnerability-scanning=disabled --no-enable-master-authorized-networks \
+        --enable-autorepair --max-surge-upgrade 1 --max-unavailable-upgrade 0 --binauthz-evaluation-mode=DISABLED \
+        --addons HorizontalPodAutoscaling,HttpLoadBalancing,GcePersistentDiskCsiDriver \
+        --no-enable-managed-prometheus 
+    return 0
+}
+
+function gke_generate_kubeconfig() {
+    local cluster_id=$1
+    local cluster_zone=$2
+    local kubeconfig_file=$3
+
+    export KUBECONFIG="${kubeconfig_file}"
+    "${GCLOUD}" container clusters get-credentials "${cluster_id}" --zone "${cluster_zone}" --project "${GCLOUD_PROJECT_ID}"
+    unset KUBECONFIG
+}
+
+FILEPATH=$(realpath "$0")
+WORKDIR=$(dirname "$FILEPATH")
+
+# shellcheck source=../../utils.sh
+source "$WORKDIR/../../utils.sh"
+
+# shellcheck source=./const.sh
+source "$WORKDIR/const.sh"
+
+RUNNER_NAME=${RUNNER_NAME:-"test"}
+CLUSTER_NAME=${RUNNER_NAME}-cluster
+
+PIDS=()
+for i in $(seq 1 "${CLUSTER_NUMBER}");
+do
+  GKE_CLUSTER_ID="${CLUSTER_NAME}${i}"
+  GKE_CLUSTER_REGION=${GKE_REGIONS[$i-1]}
+  GKE_CLUSTER_ZONE=${GKE_ZONES[$i-1]}
+
+  gke_create_cluster "${GKE_CLUSTER_ID}" "${GKE_CLUSTER_REGION}" "${GKE_CLUSTER_ZONE}" "${GKE_NUM_NODES}" "${i}" &
+  PIDS+=($!)
+done
+
+# Create GKE clusters
+for PID in "${PIDS[@]}"; do
+    wait "$PID"
+done
+
+PIDS=()
+for i in $(seq 1 "${CLUSTER_NUMBER}");
+do
+  GKE_CLUSTER_ID="${CLUSTER_NAME}${i}"
+  GKE_CLUSTER_ZONE=${GKE_ZONES[$i-1]}
+
+  gke_generate_kubeconfig "${GKE_CLUSTER_ID}" "${GKE_CLUSTER_ZONE}" "${TMPDIR}/kubeconfigs/liqo_kubeconf_${i}"
+  PIDS+=($!)
+done
+
+# Generate GKE kubeconfigs
+for PID in "${PIDS[@]}"; do
+    wait "$PID"
+done
```

Comment:
Isn't it risky to parallelize this? Consider that the gcloud function relies on an exported variable to store the kubeconfig

Reply:
It's not parallelized, but i forgot to remove the wait